Aptoide/aptoide-client-v8
# [MOB-264 - 290] Store to room and handling data migration

**What does this PR do?**

This PR aims at implementing Stores database table in Room. Also handles migrating data from the realm database to the room database, so that the not logged in users don't loose their followed stores.

**Database changed?**

   Yes

**Where should the reviewer start?**

- [ ] RoomStoreRepository.java

**How should this be manually tested?**

Open stores tab (both logged in and not logged in) and check its interactions. Follow, unfollow, open stores detail view. Test deeplinks and updates to make sure we are getting updates from followed stores.
Regarding the data migration ticket, don't forget to do a test where with a build from dev (realm database) and not logged in with several followed stores, run a build from this branch and check that the stores are still there.

**What are the relevant tickets?**

  Tickets related to this pull-request: [MOB-264](https://aptoide.atlassian.net/browse/MOB-264)

[MOB-290](https://aptoide.atlassian.net/browse/MOB-290)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
